### PR TITLE
Port enforce2d kokkos

### PR DESCRIPTION
**Pull request by Pakketeretet2**
## Purpose

This adds a port of enforce2d to Kokkos. It significantly speeds up calculations if this fix is the only non-kokkos-enabled fix so it will probably be helpful for people.

It would help if a Kokkos-expert (Stan) looks at the implementation in post_force. The calls to enforce2d of other fixes cannot assume the other fixes are Kokkos-enabled, so I figured it should be run after the parallel execution of this fix, and I also assumed it needs to sync before doing so. If there are any function calls that are not necessary, please remove them.
The current imlementation does work without issues and runs faster, but I would like to be sure. ;)
Also, the data masks don't really seem to make a difference, is that right?

## Author(s)

Stefan Paquay and Matthew Peterson @ Brandeis University

## Backward Compatibility

Just adds features, so yes.

## Implementation Notes

I just copy pasted enforce2d and aped how it works in fix_nve_kokkos

## Post Submission Checklist

_Please check the fields below as they are completed_
- [X] The feature or features in this pull request is complete
- [X] Suitable new documentation files and/or updates to the existing docs are included
- [X] The source code follows the LAMMPS formatting guidelines

### CODE REVIEW

File: src/KOKKOS/fix_enforce2d_kokkos.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  datamask_modify = V_MASK | F_MASK | OMEGA_MASK
+	  | TORQUE_MASK | ANGMOM_MASK;
+}
+
+
+template <class DeviceType>
+void FixEnforce2DKokkos<DeviceType>::setup(int vflag)
+{
+  if( comm->me == 0 ){
+    fprintf(screen, "omega, angmom and torque flags are %d, %d, %d\n",
```
Comment (stanmoore1) [inline src/KOKKOS/fix_enforce2d_kokkos.cpp]: Remove debug output

File: src/KOKKOS/fix_enforce2d_kokkos.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+struct FixEnforce2DKokkosPostForceFunctor {
+  typedef DeviceType device_type;
+  FixEnforce2DKokkos<DeviceType> c;
+
+  FixEnforce2DKokkosPostForceFunctor(FixEnforce2DKokkos<DeviceType>* c_ptr):
+    c(*c_ptr) {c.cleanup_copy();};
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i) const {
+    // c.template? Really C++?
```
Comment (stanmoore1) [inline src/KOKKOS/fix_enforce2d_kokkos.h]: Remove comment

**Comment by stanmoore1** [comment]
@Pakketeretet2 This looks pretty good, just a few tweaks requested. I'm running this through our internal Kokkos regression tests. Thanks for the contribution!

**Comment by Pakketeretet2** [comment]
That was fast! Thanks!

**Comment by stanmoore1** [comment]
@Pakketeretet2 I finally have some time to work on this...

**Comment by Pakketeretet2** [comment]
Just to be clear, I meant your response to the final changes, not the overall process; I figured you must be busy with something else. ;)

See you next week!

**Comment by stanmoore1** [comment]
@Pakketeretet2 no worries. Looks like there was a segfault from cleanup_copy in the ASPHERE examples. I switched over to `copymode` since that is the newer method of protecting the destructor.

**Comment by stanmoore1** [comment]
@Pakketeretet2 I'm going to run this through the regression tests again, and then it should be ready to merge.

**Pull request merged**